src/intfuncs.c

```diff
@@ -617,6 +617,211 @@ Obj FuncINT_STRING ( Obj self, Obj string )
 
 /****************************************************************************
 **
+*F SmallInt Bitfield operations
+*
+* The goal here it to set up a division of the usable bits in a small
+* integer into fields which can be accessed very quickly from GAP
+* level and quickly and conveniently from C. The purpose is to allow
+* implementation of data structures that divide up the bits within a
+* word without having to make them entirely opaque to the GAP level or
+* ridiculously slow
+*
+* The API is defined in lib/bitfields.gd and works by providing the
+* user with a collection of functions to get and set fields and
+* assemble an entire word.
+*
+* These functions are constructed here and have special handlers. The
+* information the handlers need about the size and position of the
+* bitfields are stored in some of the fields of the function header
+* which are not normally used for kernel functions. Specifically, we
+* use the NLOC_FUNC and FEXS_FUNC fields, which we alias as
+* MASK_BITFIELD_FUNC and OFFSET_BITFIELD_FUNC.
+*
+* For fields of size 1 we also offer Boolean setters and getters which
+* accept and return True for 1 and False for 0. This makes for much
+* nicer code on the GAP side.
+*
+*/
+
+
+static inline UInt MASK_BITFIELD_FUNC(Obj func)
+{
+  return NLOC_FUNC(func);
+}
+
+static inline void SET_MASK_BITFIELD_FUNC(Obj func, UInt mask)
+{
+  SET_NLOC_FUNC(func, mask);
+}
+
+static inline UInt OFFSET_BITFIELD_FUNC(Obj func)
+{
+  GAP_ASSERT(IS_INTOBJ(FEXS_FUNC(func)));
+  return INT_INTOBJ(FEXS_FUNC(func));
+}
+
+static inline void SET_OFFFSET_BITFIELD_FUNC(Obj func, UInt offset)
+{
+  return SET_FEXS_FUNC(func, INTOBJ_INT(offset));
+		    
+}
+
+static Obj DoFieldGetter(Obj self, Obj data)
+{
+  UInt mask = MASK_BITFIELD_FUNC(self);
+  UInt offset = OFFSET_BITFIELD_FUNC(self);
+    if (!IS_INTOBJ(data))
+        ErrorMayQuit("Field getter: argument must be small integer", 0, 0);
+    UInt x = INT_INTOBJ(data);
+    return INTOBJ_INT((x & mask)>>offset);
+}
+
+static Obj DoFieldSetter(Obj self, Obj data, Obj val)
+{
+  UInt mask = MASK_BITFIELD_FUNC(self);
+  UInt offset = OFFSET_BITFIELD_FUNC(self);
+    if (!ARE_INTOBJS(data, val))
```

Comment:
Why the differing indentation here and elsewhere?

(BTW, if you have it installed, you can use `git clang-format` to automatically format your commits)